### PR TITLE
Fix LT-19153: Unable to create fully customized writing system

### PR DESCRIPTION
**Pull request by JamesPrabu**
* Added the code to create fully costomized WS

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/721)

### CODE REVIEW

**Comment by RaymondLuong3** [comment]
*[SIL.WritingSystems/IetfLanguageTag.cs, line 63 at r1](https://reviewable.io/reviews/sillsdev/libpalaso/721#-LDsPxHR9yud_vq7CZim:-LDsPxHR9yud_vq7CZin:b-79vp35) ([raw file](https://github.com/sillsdev/libpalaso/blob/2dcd344ac4368d20f419c7a2cb46c4badfa0d55f/SIL.WritingSystems/IetfLanguageTag.cs#L63)):*
> ```Smalltalk
> 		}
> 
> 		public static bool TryGetVariantSubtags(string variantCodes, out IEnumerable<VariantSubtag> variantSubtags, string variantNames = "")
> ```

Can you leave a comment explaining why returning variantSubTags with a non-empty Name properties is important?

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/721)*

File: SIL.WritingSystems/IetfLanguageTag.cs [hunk 1]
```diff
@@ -60,7 +60,14 @@ static IetfLanguageTag()
 			PrivateUsePattern = new Regex("\\A(" + PrivateUseSubExpr + ")\\z", RegexOptions.ExplicitCapture);
 		}
 
-		public static bool TryGetVariantSubtags(string variantCodes, out IEnumerable<VariantSubtag> variantSubtags)
+		/// <summary>
+		/// This method is used to create variantSubtags based on variant code
+		/// </summary>
+		/// <param name="variantCodes">Variant code</param>
+		/// <param name="variantSubtags">return as variantsubtags</param>
+		/// <param name="variantNames">Variant name is important for creating customized writing system</param>
```
Comment (jasonleenaylor) [inline SIL.WritingSystems/IetfLanguageTag.cs]: please change the comment this way:
Change the summary to 'Return Variant Subtags based on the code, looking up and returning subtags for standard codes or creating new ones for custom (private use) codes' 
Change the comment on variantNames to 'comma separated list of variant names to use with each private use code in variantCodes'

File: SIL.WritingSystems/IetfLanguageTag.cs [hunk 2]
```diff
@@ -60,7 +60,14 @@ static IetfLanguageTag()
...
+		/// <returns></returns>
+		public static bool TryGetVariantSubtags(string variantCodes, out IEnumerable<VariantSubtag> variantSubtags, string variantNames = "")
```
Comment (jasonleenaylor) [inline SIL.WritingSystems/IetfLanguageTag.cs]: use 'string.Empty' as the default value for variantNames

**Comment by jasonleenaylor** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/721#-:-LE9dG2O98TkGcW5KVnl:bnfp4nl)*

**Pull request merged**